src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/config/AuthorizationServerConfig.kt

```diff
@@ -145,4 +151,23 @@ class AuthorizationServerConfig(
     userDetailsService.jdbcTemplate = jdbcTemplate
     return userDetailsService
   }
+
+  private fun withRequestValidatorForClientCredentials(authenticationProvider: AuthenticationProvider): AuthenticationProvider {
+    if (authenticationProvider.supports(OAuth2ClientCredentialsAuthenticationToken::class.java)) {
+      return ClientCredentialsRequestValidator(authenticationProvider, clientConfigRepository, ipAddressHelper)
+    }
+
+    return authenticationProvider
+  }
+
+  private fun OidcClientRegistrationAuthenticationProvider.getRegisteredClientConverter(): Converter<OidcClientRegistration, RegisteredClient> {
```

Comment:
Raised https://dsdmoj.atlassian.net/browse/HAAR-1556 to consider providing a full implementation of the converter, which would avoid the need for this